chainerrl/replay_buffer.py

```diff
@@ -130,19 +131,28 @@ def stop_current_episode(self):
 
 class ReplayBuffer(AbstractReplayBuffer):
 
-    def __init__(self, capacity=None):
+    def __init__(self, capacity=None, num_steps=1):
+        assert num_steps > 0
+        self.num_steps = num_steps
         self.memory = RandomAccessQueue(maxlen=capacity)
+        self.last_n_transitions = collections.deque([], maxlen=num_steps)
 
     def append(self, state, action, reward, next_state=None, next_action=None,
                is_state_terminal=False):
         experience = dict(state=state, action=action, reward=reward,
                           next_state=next_state, next_action=next_action,
                           is_state_terminal=is_state_terminal)
-        self.memory.append(experience)
+        self.last_n_transitions.append(experience)
+        if is_state_terminal:
+            self.memory.append(copy.copy(self.last_n_transitions))
```

Comment:
How about `self.memory.append(list(self.last_n_transitions))` instead of copying a deque? `self.last_n_transitions` is a deque, but transitions appended to `self.memory` do not need to be a deque.